autoprecompiles/src/memory_optimizer.rs

```diff
@@ -292,12 +251,7 @@ impl<T: FieldElement> MemoryAddressComparator<T> {
             .iter()
             .cartesian_product(b_exprs)
             .any(|(a_exprs, b_exprs)| {
-                is_value_known_to_be_different_by_word(
-                    a_exprs,
-                    b_exprs,
-                    word_size,
-                    &RangeConstraintsForBooleans,
-                )
+                is_value_known_to_be_different(a_exprs, b_exprs, &RangeConstraintsForBooleans)
```

Comment:
We could also use `a_exprs - b_exprs` here

Reply:
I meant `is_value_known_to_be_nonzero`, of course...